runsc/cmd/portforward.go

```diff
@@ -164,8 +165,16 @@ func (p *PortForward) Execute(ctx context.Context, f *flag.FlagSet, args ...any)
 }
 
 // localForward starts port forwarding from the given local port.
-func localForward(ctx context.Context, c *container.Container, localPort int, containerPort uint16) error {
-	l, err := net.Listen("tcp", ":"+strconv.Itoa(localPort))
+func localForward(ctx context.Context, c *container.Container, localSocket string, containerPort uint16) error {
+	var l net.Listener
+	localTCPPort, err := strconv.Atoi(localSocket)
+	if err == nil {
+		fmt.Printf("Forwarding local TCP port %d to %d...\n", localTCPPort, containerPort)
+		l, err = net.Listen("tcp", ":"+strconv.Itoa(localTCPPort))
+	} else {
+		fmt.Printf("Forwarding UDS at %q to %d...\n", localSocket, containerPort)
+		l, err = net.Listen("unix", localSocket)
```

Comment:
We use a UDS to send data to the sentry process. We shouldn't copy from the user provided UDS to a temporary UDS. I think we should just pass the user UDS socket directly to the sentry.

Reply:
I also thought about this and it should be possible to unify the local listener with the temporary listener (the one passed to the sentry), but afais, it will require a mutex due to possibly concurrent local connection requests. The steps of the port forwarding are as follows:

1. Accept local connection on the UDS listener
2. A Go routine for local connection passes the UDS file in a proxy forward request to the sentry
3. The UDS listener accepts sentry connection
4. Copy content between the 2 connections

If the listener in step 1 and 3 is the same, there is a risk that between 2-3, another local connection is requested. Then the listener would not accept the expected sentry connection in step 3, but another client connection. 
Do you agree or am I misunderstanding something here?

Reply:
I see. It sounds like you want to pass a UDS and support multiple connections on it whereas the sentry expects one UDS per connection request. If you want to support multiple connections on your UDS then, indeed, you'd need to map each connection there to a new temporary UDS that gets sent to the sentry and copy between them. At least as it's currently implemented.

Are you going to use this for active network requests to the sandbox? Could you just run `runsc port-forward --uds` with a different UDS for every connection? I originally implemented it this way to support port-forward for CRI via containerd, and that's generally a debugging tool. So I expected to just create and pass a different UDS per connection request.

I'm not against implementing it this way, but main thought is that I'd like to keep the interface clean such that UDS support in runsc port-forward is also used by containerd but we have not been able to get that implemented on the containerd side yet...

Reply:
oh, it looks like we already had the UDS passthrough version implemented already. So you could just use this calling `runsc port-forward --stream UDS_ADDR CONTANIER_ID REMOTE_PORT` but you'd need to create a new UDS for each connection you want to make.
https://github.com/google/gvisor/blob/master/runsc/cmd/portforward.go#L75

Reply:
Yes, we want to use it for active requests to the sandbox. So you are suggesting to open a random UDS for each request. The steps for a client request would look like this for our use case:

1. Assign a new random <UDS_ADDR>
2. Run `runsc port-forward --stream UDS_ADDR CONTAINER_ID REMOTE_PORT`
3. Client dials connection to new UDS

Is that right?

Reply:
Yes. That's what it would look like. Though I do admit it's a bit awkward for your use case.

Reply:
Thanks! We will keep the fork then for convenience and I will close the PR, since the feature is not strictly needed.